src/Symfony/Component/DependencyInjection/ServiceCollectionInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection;
+
+use Psr\Container\ContainerInterface as PsrContainerInterface;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ *
+ * @template-covariant T of mixed
+ *
+ * @extends \IteratorAggregate<string, T>
+ */
+interface ServiceCollectionInterface extends PsrContainerInterface, \IteratorAggregate, \Countable
```

Comment:
extra interface needed?

why not make ServiceLocator a container+iterator?

Reply:
> extra interface needed?

🤔 Maybe not. I was thinking it is better to type-hint/inject an interface (ie `ServiceProviderInterface`) rather than an implementation (ie `ServiceLocator`)

> why not make ServiceLocator a container+iterator?

I have no problem with this if that's the consensus.

Reply:
no need to interface value objects IMHO